cmd/daemon/main.go

```diff
@@ -109,6 +109,25 @@ const (
 	sockTCP socketType = "tcp"
 )
 
+func meshPrivateKeyCleanup(configManager config.Manager) {
+	var cfg config.Config
+	err := configManager.Load(&cfg)
+	if err != nil {
+		log.Println(internal.ErrorPrefix, "failed to load config for shutdown operations:", err)
+		return
+	}
+
+	if !cfg.Mesh && cfg.MeshPrivateKey != "" {
+		err := configManager.SaveWith(func(c config.Config) config.Config {
+			c.MeshPrivateKey = ""
+			return c
+		})
+		if err != nil {
+			log.Println(internal.ErrorPrefix, "failed to clean up mesh private key on shutdown:", err)
+		}
```

Comment:
Would it make sense to move this to separate method? It's almost the same as the changes in `rpc_disconnect.go`, only log message differs